frontend/src/app/testQueue/TestCardForm/diseaseSpecificComponents/TestResultInputGroup.tsx

```diff
@@ -0,0 +1,123 @@
+import { Alert } from "@trussworks/react-uswds";
+
+import {
+  MultiplexResultInput,
+  SupportedDiseaseTestPerformed,
+} from "../../../../generated/graphql";
+import { DevicesMap } from "../../QueueItem";
+import RadioGroup, {
+  RadioGroupOptions,
+} from "../../../commonComponents/RadioGroup";
+import { TEST_RESULTS } from "../../../testResults/constants";
+import { TEST_RESULT_DESCRIPTIONS } from "../../../constants";
+
+interface Props {
+  testOrderId: string;
+  testResults: MultiplexResultInput[];
+  deviceId: string;
+  devicesMap: DevicesMap;
+  onChange: (value: MultiplexResultInput[]) => void;
+}
+
+type TestResultRecord = Record<string, TEST_RESULTS>;
+
+const convertTestResultRecordToArray = (testResultRecord: TestResultRecord) => {
+  return Object.entries(testResultRecord).map((entry): MultiplexResultInput => {
+    return {
+      diseaseName: entry[0],
+      testResult: entry[1],
+    };
+  });
+};
+
+const convertMultiplexResultArrayToRecord = (
+  testResults: MultiplexResultInput[]
+) => {
+  const testResultRecords: TestResultRecord = {};
+  testResults.forEach((value) => {
+    if (value.diseaseName) {
+      testResultRecords[value.diseaseName] =
+        (value.testResult as TEST_RESULTS) ?? TEST_RESULTS.UNKNOWN;
+    }
+  });
+  return testResultRecords;
+};
+
+export const TestResultInputGroup = ({
+  testOrderId,
+  testResults,
+  deviceId,
+  devicesMap,
+  onChange,
+}: Props) => {
+  const supportedDiseaseTests =
```

Comment:
Thanks for finding this case! We might be able to use `test performed code` to determine whether the diseases are identical and filter it out of the list, but I'm checking with product on whether that's a reliable field to use.

I don't think it'll necessarily matter which COVID we use in this case since the test card just sends the disease name and test result along with the selected device instead of sending anything specific to the `test performed code` or `test ordered code`

Reply:
Just pushed a commit that should filter this out based on the duplicate disease name